embeddings/config/flair_config.py

```diff
@@ -0,0 +1,174 @@
+from abc import ABC
+from dataclasses import dataclass, field
+from itertools import chain
+from types import MappingProxyType
+from typing import Any, ClassVar, Dict, Mapping, Set, Tuple, Union
+
+from embeddings.config.base_config import AdvancedConfig, BasicConfig
+from embeddings.data.io import T_path
+from embeddings.utils.utils import read_yaml
+
+
+@dataclass
+class FlairTextClassificationConfigMapping:
+    LOAD_MODEL_KEYS_MAPPING: ClassVar[Mapping[str, Set[str]]] = MappingProxyType(
+        {
+            "FlairDocumentCNNEmbeddings": {
+                "hidden_size",
+                "rnn_type",
+                "rnn_layers",
+                "bidirectional",
+                "dropout",
+                "word_dropout",
+                "reproject_words",
+            },
+            "FlairDocumentRNNEmbeddings": {
+                "cnn_pool_kernels",
+                "dropout",
+                "word_dropout",
+                "reproject_words",
+            },
+            "FlairTransformerDocumentEmbedding": {"pooling", "fine_tune"},
+            "FlairDocumentPoolEmbedding": {"pooling", "fine_tune_mode"},
+        }
+    )
+
+    @classmethod
+    def map_load_model_keys(cls, document_embedding_cls: str) -> Set[str]:
+        return cls.LOAD_MODEL_KEYS_MAPPING[document_embedding_cls]
+
+    @classmethod
+    def get_load_model_keys(cls) -> Set[str]:
+        return set(chain(*cls.LOAD_MODEL_KEYS_MAPPING.values()))
+
+
+@dataclass
+class FlairBasicConfig(BasicConfig):
+    learning_rate: float = 1e-3
+    mini_batch_size: int = 32
+    max_epochs: int = 20
+
+    task_model_kwargs: Dict[str, Any] = field(init=False, compare=False, default_factory=dict)
+    task_train_kwargs: Dict[str, Any] = field(init=False, compare=False, default_factory=dict)
```

Comment:
we do not need to include keyword arguments in comparison methods since these are populated using init arguments; however, it does not change the results of the comparison since these dictionaries will also be equal as they do not store any complex objects